### ISSUE TITLE
Suggestion: Choose Promise library that doesn't depend on nextTick

### ISSUE THREAD

**Issue by dwighthouse** [pyrsmk/qwest#92]
Great project, but I have an issue when using this in a browser.

When using qwest, it is impossible to avoid the inclusion of a [process shim by browserify](https://github.com/defunctzombie/node-process) which emulates the node-native `process` object. Particularly, it adds the `nextTick` functionality, which PinkySwear (the Promise library used by qwest) uses as a fallback when available. Including this shim adds a completely unnecessary 1.3KB to the JS after minification (2KB before minification).

I have been using a transform called [envify](https://github.com/hughsk/envify/) to remove instances of `process` from my code, but that only works for environmental variables. It can't refactor libraries.

I would suggest a different Promise library, but I don't tend to use Promises in my own code myself, so I don't know which to suggest. That, or perhaps a fork of PinkySwear that doesn't require process. Thanks for your consideration.

**Comment by pyrsmk** [comment]
Ah, thanks, I didn't now that.

But, actually, I don't exactly understand how it works : when browserify encounters a call to `process.nextTick()`, is it automatically adding the shim? Or browserify is adding it in all cases?

**Comment by dwighthouse** [comment]
Since `process` is a node-native object that doesn't exist in the browser, Browserify adds the shim if it encounters any usage of the `process` object. Normally, the only usage of process in code aimed at the browser is to allow for environmental variables. For example, when one does this:
```
if (process.env.NODE_ENV !== 'production')
{
    // debug code
}
```
One can set the NODE_ENV to 'development' or 'production' for different builds. When the code runs through `envify` using 'production', it becomes:
```
if ('production' !== 'production')
{
    // debug code
}
```
The contents of the if-block will never be reached. A good minifier, like UglifyJS2, will strip these blocks. This is why the production and debug builds of ReactJS are so different in size and runtime performance. All the debug warnings, error messages, checks, and some performance analysis code are being stripped.

`PinkySwear`, however, uses `process`, specifically it's `nextTick` function. `nextTick` is a function, not an environmental variable, so there's no way to replace it, and so the `process` shim gets included anyway. As described in [Browserify's Compatibility section](https://github.com/substack/node-browserify#compatibility),

> Additionally, if you use any of these variables, they will be defined in the bundled output in a browser-appropriate way:

> process
> Buffer
> global - top-level scope object (window)
> __filename - file path of the currently executing file
> __dirname - directory path of the currently executing file

**Comment by pyrsmk** [comment]
Thanks for the explanations!

So... A good fix would be first replacing `process` in PinkySwear by anything else; like `derequire` does (if I'm right). Then we can bundle the whole thing and browserify won't add its shim.

**Comment by dwighthouse** [comment]
That's not really possible because Browserify is the entry point for JS, so there would be no way to process the files prior to Browserify getting them.

After a LOT of trial and error, I found that adding this to my configuration object would successfully prevent the inclusion of `process` AND prevent Browserify from attempting to require it.

```
browserify({
...,
insertGlobalVars: {
    process: false
},
...
})
```

Looks simple, but I tried about 6 different ways to prevent it, and this one was the first one that worked fully.
